src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -401,7 +405,14 @@ export function loadOperation(
                 StatusCodes: op.verb === "delete" ? [204] : [200],
                 BodyType: bodyType,
                 BodyMediaType: BodyMediaType.Json
-            } as OperationResponse
+            } as OperationResponse,
+            ResultPath:
+                metadata.logicalPath ??
+                // https://github.com/Azure/typespec-azure/issues/3725
```

Comment:
Add a TODO here to inform we will remove this workaround after issue 3725 is resoved.